cfspopcon/formulas/scrape_off_layer/lambda_q.py

```diff
@@ -42,58 +44,87 @@ def calc_lambda_q(
         :term:`lambda_q` [mm]
     """
     if lambda_q_scaling == LambdaQScaling.Brunner:
-        return lambda_q_factor * float(calc_lambda_q_with_brunner.unitless_func(average_total_pressure))
+        return float(
+            calc_lambda_q_with_brunner.unitless_func(average_total_pressure=average_total_pressure, lambda_q_factor=lambda_q_factor)
+        )
     elif lambda_q_scaling == LambdaQScaling.EichRegression14:
-        return lambda_q_factor * float(calc_lambda_q_with_eich_regression_14.unitless_func(B_pol_out_mid))
+        return float(calc_lambda_q_with_eich_regression_14.unitless_func(B_pol_out_mid=B_pol_out_mid, lambda_q_factor=lambda_q_factor))
     elif lambda_q_scaling == LambdaQScaling.EichRegression15:
-        return lambda_q_factor * float(
+        return float(
             calc_lambda_q_with_eich_regression_15.unitless_func(
-                power_crossing_separatrix, major_radius, B_pol_out_mid, inverse_aspect_ratio
+                power_crossing_separatrix=power_crossing_separatrix,
+                major_radius=major_radius,
+                B_pol_out_mid=B_pol_out_mid,
+                inverse_aspect_ratio=inverse_aspect_ratio,
+                lambda_q_factor=lambda_q_factor,
             )
         )
     else:
         raise NotImplementedError(f"No implementation for lambda_q scaling {lambda_q_scaling}")
 
 
+@Algorithm.register_algorithm(return_keys=["lambda_q"])
 @wraps_ufunc(
     return_units=dict(lambda_q=ureg.millimeter),
-    input_units=dict(average_total_pressure=ureg.atm),
+    input_units=dict(average_total_pressure=ureg.atm, lambda_q_factor=ureg.dimensionless),
 )
-def calc_lambda_q_with_brunner(average_total_pressure: float) -> float:
+def calc_lambda_q_with_brunner(average_total_pressure: float, lambda_q_factor: float = 1.0) -> float:
```

Comment:
What is the motivation to move this multiplicative factor inside the function?

Reply:
Needed to prepare for removing calc_lambda_q